dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/gdv/custom/GdvListOfBaseDataPointComponent.kt

```diff
@@ -0,0 +1,82 @@
+package org.dataland.frameworktoolbox.frameworks.gdv.custom
+
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.addStandardCellWithValueGetterFactory
+import org.dataland.frameworktoolbox.intermediate.datapoints.SimpleDocumentSupport
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.SectionUploadConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+
+/**
+ * A GdvYearlyDecimalTimeseriesDataComponent is an in-memory representation of a generic field
+ * that encodes several values across a span of multi years. It is displayed / upload in a matrix
+ */
+class GdvListOfBaseDataPointComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    lateinit var descriptionColumnHeader: String
+    lateinit var documentColumnHeader: String
```

Comment:
`lateinit` vars can cause `NullPointerException`. Use default value / nullable type instead and throw an error when values are still null during code generation.

Reply:
What is the advantage of doing that ?   Feel free to call me and explain it. Interesting stuff

Reply:
Were you able to clarify this during your call?

Reply:
Yes. We will use "requireNotNull" on these fields when we use this component.
Then they can be normal class variables here.

Reply:
Done.

- Removed the "lateinit" => the variables are "normal" vars now.
- Used a "requireNotNull" on the two variables before using them in `generateDefaultViewConfig()`